frontend/src/pages/team/Devices/dialogs/DeviceCredentialsDialog.vue

```diff
@@ -69,6 +75,19 @@ export default {
         close () {
             this.$refs.dialog.close()
             this.device.credentials = undefined
+        },
+        copyToClipboard () {
+            navigator.permissions.query({ name: 'clipboard-write' }).then((result) => {
+                if (result.state === 'granted' || result.state === 'prompt') {
+                    /* write to the clipboard now */
+                    navigator.clipboard.writeText(this.credentials)
+                    Alerts.emit('Copied to Clipboard.', 'confirmation')
+                }
+            }).catch((err) => {
+                // eslint-disable-next-line no-console
+                console.log('Clipboard write permission denied: ', err)
```

Comment:
Use console.warn here and you can remove the eslint disable

Reply:
I chose log as it's not really an issue and didn't want an ugly warn in the console. I'm happy to go with warn if that is our preferred means of putting a debug statement in the console?  

Thoughts?

Reply:
Hard to say, and admittedly our guidelines aren't written yet. I'd lean towards not logging anything at all if it's not worthy of a warning or error.

In this case I do think denying the keyboard permissions is worthy of a warning. It's not an error state, it's not an info, it's somewhere in the middle.

Reply:
resolved in commit [4b00087](https://github.com/flowforge/flowforge/pull/2231/commits/4b0008762fa7960e821c37fcaf0f8f177883e774)